workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/src/Generation/Model/Details/UnityFieldDetails.cs

```diff
@@ -1,44 +1,68 @@
-using System.Collections.Generic;
-using Improbable.CodeGeneration.Model;
-using Improbable.CodeGeneration.Utils;
+using System;
+using System.Collections.ObjectModel;
+using Improbable.Gdk.CodeGeneration.Model.SchemaBundleV1;
+using Improbable.Gdk.CodeGeneration.Utils;
 
 namespace Improbable.Gdk.CodeGenerator
 {
     public class UnityFieldDetails
     {
         public string Type => fieldType.Type;
-        public string PascalCaseName => Formatting.SnakeCaseToCapitalisedCamelCase(rawFieldDefinition.name);
-        public string CamelCaseName => Formatting.SnakeCaseToCamelCase(rawFieldDefinition.name);
-        public int FieldNumber => rawFieldDefinition.Number;
-        public bool IsBlittable;
-        public bool CanBeEmpty;
+
+        public string PascalCaseName { get; }
+        public string CamelCaseName { get; }
+        public uint FieldNumber { get; }
+
+        public bool IsBlittable { get; }
+        public bool CanBeEmpty { get; }
 
         private IFieldType fieldType;
-        private FieldDefinitionRaw rawFieldDefinition;
 
-        public UnityFieldDetails(FieldDefinitionRaw rawFieldDefinition, bool isBlittable, HashSet<string> enumSet)
+        public UnityFieldDetails(Field field, DetailsStore store)
         {
-            this.rawFieldDefinition = rawFieldDefinition;
-            IsBlittable = isBlittable;
+            PascalCaseName = Formatting.SnakeCaseToPascalCase(field.Identifier.Name);
+            CamelCaseName = Formatting.PascalCaseToCamelCase(PascalCaseName);
+            FieldNumber = field.FieldId;
 
-            if (rawFieldDefinition.IsOption())
+            if (field.Option != null)
             {
-                fieldType = new OptionFieldType(rawFieldDefinition.optionType, enumSet);
+                IsBlittable = false;
```

Comment:
It feels like we're duplicating the code for blittable check here. Any way we can merge them into a single function?

Reply:
Hmm maybe.. the problem I'm having is that the methods in `DetailsStore` are used in the context of - "we are building a blittable map for all identifiers" vs "we are doing a lookup in that completed map". 

So for things like option, map, list, that's fine - but when you go to check the `UserType` of a singular type, you end up accessing a different data type (a mutable dictionary in the `DetailsStore` internals, and an `ImmutableHashSet` in the `FieldDetails`).

There is likely a way to factor this out nicely - probably by using the `Identifier` of the `Field` to make entries into the `ImmutableHashSet` so its always just a lookup in the `UnityFieldDetails`

Reply:
Can we have one function that takes a Map/HashSet of already identified Types and just pass incomplete or complete one depending on the context where it is called?

Reply:
Refactored this in https://github.com/spatialos/gdk-for-unity/pull/737/commits/89f563292bca13d6b32a787649e8d765e3e7699d PTAL 😄

Reply:
Love it!